src/backtrace.c

```diff
@@ -108,8 +108,10 @@ int backtrace_snapshot(int pid, int *tids, int *index, int nr_tids)
         return -1;
 
     for (i = 0; i < snap->num_threads; ++i) {
-        printf("--------------------  thread %d (%d)  --------------------\n",
+        printf("--------------------  thread %d (%d) (",
                (index != NULL ? index[i] : i+1), snap->tids[i]);
+        print_proc_comm(snap->tids[i]);
+        printf(")  --------------------\n");
```

Comment:
I agree, no need to store it because we use it single time at this point.

We can have 80+1 character buffer filled up with '-', then use `snprintf()` to print thread number, pid, and name. We could make `print_proc_comm()` returning pointer to static data so only single `snprintf()` will be needed. Not sure it is best/simplest solution, it's first coming to my mind. Or better we can prepare the title in another buffer, calculate its length and choose a position in "----" to make it center aligned.